pyscriptjs/tests/integration/test_plugins.py

```diff
@@ -226,13 +226,18 @@ def test_execution_hooks(self):
 
         # EXPECT it to log the correct logs for the events it intercepts
         log_lines = self.console.log.lines
-        for method in hooks_available:
-            assert log_lines.count(f"{method} called") == 1
+        num_calls = {
+            method: log_lines.count(f"{method} called") for method in hooks_available
+        }
+        expected_calls = {method: 1 for method in hooks_available}
+        assert num_calls == expected_calls
```

Comment:
@hoodmane suggested it to do it this way

Reply:
This makes much more detailed assertion errors when the test fails.

Reply:
To be clear, the old logic also didn't test order. I think:
1. The new check fails if and only if the old check fails
2. When the new check fails, the error message pytest formats includes a super set of the information in the old message.

In general, 
```py
for x in it:
   assert f(x) == g(x)
```
Should be converted to something like
```py
result = []
expected = []
for x in it:
   result.append([x, f(x)])
   expected.append([x, g(x)])
assert result == expected
```
For better assertion errors. (Incidentally, this is a variant of the same code transform where you switch from `await` in a loop to using `Promise.all`...)

Reply:
> In general,
> 
> ```python
> for x in it:
>    assert f(x) == g(x)
> ```
> 
> Should be converted to something like
[cut]
 
> For better assertion errors.

I disagree:
1. with the first variant, it's obvious what's going on. With the 2nd variant, I need to understand what is the testing logic (and complex testing logic is prone to errors)
2. The fist variant is much easier to debug locally, because you can just use `--pdb` and interactively check what's going on. With the second variant, this is impossible because the assert fails much later.

That said, I don't care enough to block the merging because of this, so I'm marking this as resolved